lib/carto/tree/filterset.js

```diff
@@ -93,9 +93,9 @@ tree.Filterset.prototype.toJS = function(env) {
     }
     var attrs = "data";
     if (op === '=~') {
-      return "(" + attrs + "['" + filter.key.value  + "'] + '').match(" + (val.is === 'string' ? "'" + val.toString().replace(/'/g, "\\'") + "'" : val) + ")";
+      return "(" + attrs + "['" + filter.key.value  + "'] + '').match(" + (val.is === 'string' ? "'" + val.toString().replace(/'/g, "\\'").replace(/&amp;/g, '&') + "'" : val) + ")";
```

Comment:
The added replace calls look good. But the whole expressions are unintelligible️ :S 
But this should go in another issue.